src/allmydata/test/cli/test_cp.py

```diff
@@ -238,6 +238,70 @@ def _check(args):
 
         return d
 
+    @defer.inlineCallbacks
+    def test_cp_duplicate_directories(self):
+        self.basedir = "cli/Cp/cp_duplicate_directories"
+        self.set_up_grid(oneshare=True)
+
+        filename = os.path.join(self.basedir, "file")
+        data = b"abc\xff\x00\xee"
+        with open(filename, "wb") as f:
+            f.write(data)
+
+        yield self.do_cli("create-alias", "tahoe")
+        (rc, out, err) = yield self.do_cli("mkdir", "tahoe:test1")
+        self.assertEqual(rc, 0, (rc, err))
+        dircap = out.strip()
+
+        (rc, out, err) = yield self.do_cli("cp", filename, "tahoe:test1/file")
+        self.assertEqual(rc, 0, (rc, err))
+
+        # Now duplicate dirnode, testing duplicates on destination side:
+        (rc, out, err) = yield self.do_cli(
+            "cp", "--recursive", dircap, "tahoe:test2/")
+        self.assertEqual(rc, 0, (rc, err))
+        (rc, out, err) = yield self.do_cli(
+            "cp", "--recursive", dircap, "tahoe:test3/")
+        self.assertEqual(rc, 0, (rc, err))
+
+        (rc, out, err) = yield self.do_cli("ls", "tahoe:test1")
+        (rc, out, err) = yield self.do_cli("ls", "tahoe:test2")
+        (rc, out, err) = yield self.do_cli("ls", "tahoe:test3")
```

Comment:
What are these `tahoe ls` lines for?  Can we document them if they have a purpose, or remove them if they do not?

Reply:
Hm. That might've just been some debugging code. I'll delete.